packages/core/src/sequelize.js

```diff
@@ -873,6 +872,8 @@ Use Sequelize#query if you wish to use replacements.`);
    */
   async _syncModelsWithCyclicReferences(options) {
     if (this.dialect.name === 'sqlite') {
+      // eslint-disable-next-line import/no-extraneous-dependencies -- only import when needed
+      const { withSqliteForeignKeysOff } = await import('@sequelize/sqlite');
```

Comment:
This is not ideal however if the `@sequelize/sqlite` package becomes a dependency, we end up with having cyclic dependencies.

Reply:
We can move this function to the core utils folder for now and figure out a better way later

Reply:
Updated now.